api/v1/cluster_types.go

```diff
@@ -1462,30 +1462,40 @@ func (cluster *Cluster) GetApplicationSecretName() string {
 
 // GetApplicationDatabaseName get the name of the application database for a specific bootstrap
 func (cluster *Cluster) GetApplicationDatabaseName() string {
-	switch {
-	case cluster.ShouldRecoveryCreateApplicationDatabase():
-		return cluster.Spec.Bootstrap.Recovery.Database
-	case cluster.ShouldPgBaseBackupCreateApplicationDatabase():
-		return cluster.Spec.Bootstrap.PgBaseBackup.Database
-	case cluster.ShouldInitDBCreateApplicationDatabase():
-		return cluster.Spec.Bootstrap.InitDB.Database
-	default:
-		return ""
+	bootstrap := cluster.Spec.Bootstrap
```

Comment:
Bootstrap could be `nil` here. It is unlikely, but it is worth adding an if for protection.